src/record_signal.cc

```diff
@@ -166,9 +166,14 @@ static bool try_grow_map(RecordTask* t, siginfo_t* si) {
     LOG(debug) << "try_grow_map " << addr << ": address would be in guard page";
     return false;
   }
-  struct rlimit64 stack_limit;
   remote_ptr<void> limit_bottom;
+#if not defined (__i386__)
```

Comment:
Use the `!` operator please ... I didn't even know you can write `not` here

Reply:
Actually, flip the order of the if/else blocks so you don't need to use "not" at all.